doc/tools/docgram/orderedGrammar

```diff
@@ -1742,6 +1742,8 @@ simple_tactic: [
 | "autounfold" OPT hintbases OPT simple_occurrences
 | "autounfold_one" OPT hintbases OPT ( "in" ident )
 | "unify" one_term one_term OPT ( "with" ident )
+| "typeclasses" "eauto" "dfs" OPT nat_or_var "with" LIST1 ident
+| "typeclasses" "eauto" "dfs" OPT nat_or_var
```

Comment:
For the doc, these added lines should merged into the existing syntax for the tactic by editing `common.edit_mlg` like this (not tested):

Replace

```
| "typeclasses" "eauto" OPT "bfs" OPT nat_or_var OPT ( "with" LIST1 preident )
```

with

```
| "typeclasses" "eauto" OPT ( "bfs" | "dfs ) OPT nat_or_var OPT ( "with" LIST1 preident )
| DELETE "typeclasses" "eauto" "dfs" OPT nat_or_var "with" LIST1 preident
| DELETE "typeclasses" "eauto" "dfs" OPT nat_or_var
```

then run `make doc_gram_rsts`, which will update the syntax in the rst files.

Also if you would describe both bfs and dfs in the tactic description just above these lines (see the `Typeclasses eauto` command description of these) and with the same format (e.g. ```:n:`{| bfs | dfs }```):

```
   :n:`@nat_or_var`
     Specifies the maximum depth of the search.
```